common/src/main/java/whocraft/tardis_refined/common/world/chunk/TardisChunkGenerator.java

```diff
@@ -23,26 +22,124 @@
 import net.minecraft.world.level.levelgen.Heightmap;
 import net.minecraft.world.level.levelgen.RandomState;
 import net.minecraft.world.level.levelgen.blending.Blender;
+import net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings;
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager;
+import whocraft.tardis_refined.TardisRefined;
 import whocraft.tardis_refined.common.world.ChunkGenerators;
-import whocraft.tardis_refined.constants.TardisGeneration;
+import whocraft.tardis_refined.constants.TardisDimensionConstants;
+import whocraft.tardis_refined.registry.ARSStructurePieceRegistry;
 import whocraft.tardis_refined.registry.BlockRegistry;
 
 import java.util.List;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 public class TardisChunkGenerator extends ChunkGenerator {
-    public static final Codec<TardisChunkGenerator> CODEC = RecordCodecBuilder.create(instance ->
-            instance.group(RegistryOps.retrieveElement(ChunkGenerators.TARDIS_BIOME)).apply(instance,
-                    instance.stable(TardisChunkGenerator::new)));
+    public static final Codec<TardisChunkGenerator> CODEC = RecordCodecBuilder.create(instance -> instance.group(RegistryOps.retrieveElement(ChunkGenerators.TARDIS_BIOME)).apply(instance, instance.stable(TardisChunkGenerator::new)));
+
+
+    public final RandomSource random;
+
+    // Some parameter values.
+    public final int distanceBetweenGravityWell = 18;
+    public final int arsChunkSize = ARSStructurePiece.LOCKED_PIECE_CHUNK_SIZE;
+    private final int chunkSize = 16;
+
 
     public TardisChunkGenerator(Holder<Biome> holder) {
         super(new FixedBiomeSource(holder));
+        this.random = RandomSource.create();
     }
 
     @Override
     public void applyBiomeDecoration(WorldGenLevel pLevel, ChunkAccess pChunk, StructureManager pStructureManager) {
+
+        if (pChunk.getPos().x == 63 && pChunk.getPos().z == 0) {
+            pLevel.getLevel().getStructureManager().get(new ResourceLocation(TardisRefined.MODID, "corridors/corridor_hub/corridor_hub_center")).ifPresent(structure -> {
```

Comment:
These ResourceLocations can be constants in the ARSStructurePiece registry class so that we aren't creating the ResourceLocation on the fly.